lib/topological_inventory/orchestrator/worker.rb

```diff
@@ -111,21 +115,31 @@ def url_for(path)
         File.join(api_base_url, path)
       end
 
-      def each_resource(url, &block)
+      def each_resource(url, tenant_account = ORCHESTRATOR_TENANT, &block)
         return if url.nil?
-        response = get_and_parse(url)
+        response = get_and_parse(url, tenant_account)
         paging = response.is_a?(Hash)
 
         resources = paging ? response["data"] : response
         resources.each { |i| yield i }
 
-        each_resource(response["links"]["next"], &block) if paging
+        each_resource(response["links"]["next"], tenant_account, &block) if paging
       end
 
-      def get_and_parse(url)
-        JSON.parse(RestClient.get(url))
+      def get_and_parse(url, tenant_account = ORCHESTRATOR_TENANT)
+        JSON.parse(
+          RestClient.get(
+            url,
+            "x-rh-identity" => Base64.encode64(
+              {"identity" => {"account_number" => tenant_account}}.to_json
+            ).strip
```

Comment:
This feels like something we shouldn't do here, but I'm ok temporarily.  I'd expect a) building identity would live in common, so we are not duplicating this common thing and b) we wouldn't use RestClient, but a proper client library.

Reply:
Opened https://github.com/ManageIQ/topological_inventory-orchestrator/issues/11 for this